tests/test_ec2.py

```diff
@@ -502,3 +502,126 @@ def test_action_construction(self):
+
+        default_sg_id = client.describe_security_groups(
+            GroupNames=[
+                'default',
+            ]
+        )['SecurityGroups'][0]['GroupId']
+
+
+        # Catch on anything that uses the *PROD-ONLY* security groups but isn't in a prod role
+        policy = self.load_policy({
+            'name': 'remove-sensitive-sg',
+            'resource': 'ec2',
+            'filters': [
+                {'or': [
+                    {'and': [
+                        {'type': 'value', 'key': 'IamInstanceProfile.Arn', 'value': '(?!.*TestProductionInstanceProfile)(.*)', 'op': 'regex'},
+                        {'type': 'value', 'key': 'IamInstanceProfile.Arn', 'value': 'not-null'}
+                    ]},
+                    {'type': 'value', 'key': 'IamInstanceProfile', 'value': 'absent'}
+                ]},
+                {'type': 'security-group', 'key': 'GroupName', 'value': '(.*PROD-ONLY.*)', 'op': 'regex'}],
+            'actions': [
+                {'type': 'modify-security-groups', 'remove': 'matched', 'isolation-group': default_sg_id}]
+            },
+            session_factory=session_factory)
+        before_action_resources = policy.run()
+        after_action_resources = policy.run()
+        self.assertEqual(len(before_action_resources), 1)
+        self.assertEqual(before_action_resources[0]['InstanceId'], 'i-0dd3919bc5bac1ea8')
+        self.assertEqual(len(after_action_resources), 0)
+
+
+    def test_invalid_modify_groups_schema(self):
+        session_factory = self.replay_flight_data(
+            'test_ec2_invalid_modify_groups_schema'
+        )
+
+        class foo():
+            def __init__(self, test_obj):
+                self.test_obj = test_obj
+            def bar(self):
+                policy = self.test_obj.load_policy({
+                    'name': 'invalid-modify-security-groups-action',
+                    'resource': 'ec2',
+                    'filters': [
+                        {'or': [
+                            {'and': [
+                                {'type': 'value', 'key': 'IamInstanceProfile.Arn', 'value': '(?!.*TestProductionInstanceProfile)(.*)', 'op': 'regex'},
+                                {'type': 'value', 'key': 'IamInstanceProfile.Arn', 'value': 'not-null'}
+                            ]},
+                            {'type': 'value', 'key': 'IamInstanceProfile', 'value': 'absent'}
+                        ]},
+                        {'type': 'security-group', 'key': 'GroupName', 'value': '(.*PROD-ONLY.*)', 'op': 'regex'}],
+                    'actions': [
+                        {'type': 'modify-security-groups', 'change': 'matched'}
+                    ]
+                },
+                session_factory=session_factory)
+                resources = policy.run()
+                return resources
+
+        i = foo(self)
+        self.assertRaises(ValueError, lambda: i.bar())
+
+
+    def test_ec2_add_security_groups(self):
```

Comment:
for tests with extant state, could you add a # comments on the preconditions the test is expecting.